### PR TITLE
Multiple Formats in a Text Field

### LINKED ISSUE #985: Multiple Label Styles with Format Expression
Given that https://github.com/mapbox/mapbox-gl-native/pull/12985 has landed, it's now possible to style a SymbolLayer's text field with multiple fonts, sizes, etc. It would be great to have an example in the app that showcased this (with a bit more polish than https://github.com/mapbox/mapbox-gl-native/pull/12985#issuecomment-469643033). Something as simple as [the example that was included in the gl-native test app](https://github.com/mapbox/mapbox-gl-native/blob/8a8291d8f4ce39d341f42c6d14b2aaad28cd854e/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/style/SymbolLayerActivity.java#L127-L136) should do just fine.

Note that this is a slightly different example than https://github.com/mapbox/mapbox-android-demo/pull/982 as it will have multiple styles within the textField. Though this would be complimentary to that, it shows off a different possibility for the expression. 

cc: @langsmith @chloekraw

### PR DESCRIPTION
**Pull request by riastrad**
Resolves #985 

This PR adds a new example that demonstrates how to to use multiple formats within a `SymbolLayer`'s textField.

### CODE REVIEW

**Comment by langsmith** [comment]
@riastrad , https://docs.mapbox.com/mapbox-gl-js/example/display-and-style-rich-text-labels/ could be good to reference if you wanna' see how the expressions are used. All of these wild text label examples are basically more about setting default, fallbacks, etc. with expressions, than anything else.

**Comment by riastrad** [comment]
@langsmith this basically done (let's say 98%). I'll be updating the example's menu card image tomorrow morning and then it should be good to merge 🚀

multiple-formats-screenshot

**Comment by langsmith** [comment]
giphy

**Comment by riastrad** [comment]
This should be good to go, @langsmith! 

Also noting that I dropped this in a very similar location as the new example in https://github.com/mapbox/mapbox-android-demo/pull/982, so whichever example merges second should gird itself for the probability of merge errors in `MainActivity.java` & `AndroidManifest.xml`.

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/TextFieldMultipleFormatsActivity.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.maps.Style;
+import com.mapbox.mapboxsdk.style.expressions.Expression;
+import com.mapbox.mapboxsdk.style.layers.Layer;
+
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.textField;
+
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/TextFieldMultipleFormatsActivity.java]: Add `/**` javadocs to describe what this example is about

```
/**
 * Description of what this example shows
 */
public class TextFieldMultipleFormatsActivity extends AppCompatActivity {
```

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/TextFieldMultipleFormatsActivity.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+              get("name"),
+              formatTextFont(new String[] {"Caveat Regular", "Arial Unicode MS Regular"})
+            );
+
+            Expression format = format(bigLabel, newLine, smallLabel);
+
+            // Retrieve the country label layers from the style and update them with the formatting expression
+            for (Layer mapLabelLayer : style.getLayers()) {
+              if (mapLabelLayer.getId().contains("country-label")) {
+                // apply formatting expression
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/TextFieldMultipleFormatsActivity.java]: Capitalize `apply` 😬 

```suggestion
                // Apply formatting expression
```

**Comment by langsmith** [comment]
Great work @riastrad . I pushed a commit, which imports some methods to improve the readability of the code, so that there are fewer instances of `Expression.` throughout the code.

ezgif com-resize (3)

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/TextFieldMultipleFormatsActivity.java (hunk 2 above)
Comment (riastrad) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/TextFieldMultipleFormatsActivity.java]: 😐

**Comment by riastrad** [comment]
Thanks for the review, @langsmith. This is ready for another quick pass as soon as you're ready.

**Comment by langsmith** [comment]
Looks good!

**Pull request merged**